### ISSUE TITLE
What is the time schedule for ASP.NET Core 2.0 support

### ISSUE THREAD

**Issue by dozer75** [aspnet-contrib/AspNet.Security.OpenIdConnect.Server#478]
This repository is quite confusing, it refers to Core 2.0 support, but the libraries on NuGet is still 1.x libraries.  And to be able to upgrade to asp.net core 2.0 (which I need to get EF to work) I need this to work to.

And also, will there be any real documentation on how to upgrade this?

**Comment by DovydasNavickas** [comment]
https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/issues/473
You might not be using the preview version 😉

**Comment by kevinchalet** [comment]
> This repository is quite confusing, it refers to Core 2.0 support, but the libraries on NuGet is still 1.x libraries. 

The 2.0 nightly builds are available on the MyGet feed mentioned in the README. A first public version will be pushed to NuGet.org soon.

> And also, will there be any real documentation on how to upgrade this?

Almost all the changes made as part of this release are to make it compatible with the new ASP.NET Core authentication stack, whose migration process is already heavily documented: https://github.com/aspnet/Announcements/issues/262.

I don't plan to write anything at this point, as there's nothing ASOS-specific (unless someone wants to finally support the task, of course).

**Comment by dozer75** [comment]
@DovydasNavickas I am using the preview version, this was a general question since I am in a production environment and don't want to use previews.

@PinpointTownes Thanks for answers. Yes, I am using the nightly builds, however, there are things that has changed and not documented and this is a bit frustrating.

I really don't like that the general authentication configuration has been moved to service provider configuration in ASP.NET Core 2.0. In my call to UseOpenIdConnectServer in 1.0 I am using the service provider to retrieve logging, but this isn't possible now since the logging isn't available (as far as I can find anyway), and I have no idea how to solve this. And in my OpenIdConnectServerProvider override, I am heavily dependent on dependency injection, but due to the current implementation I have no idea how to retrieve this. 

In addition the previous implemlentation of OpenIdConnectServerOptions contained AuthenticationScheme which isn't available anymore and no documentation on how to retrieve it.

**Comment by kevinchalet** [comment]
> And in my OpenIdConnectServerProvider override, I am heavily dependent on dependency injection, but due to the current implementation I have no idea how to retrieve this.

Did you take a look at the sample? Your provider can now be registered as a scoped dependency in DI and resolved at request time. https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/dev/samples/Mvc/Mvc.Server/Startup.cs#L52

> In addition the previous implemlentation of OpenIdConnectServerOptions contained AuthenticationScheme which isn't available anymore and no documentation on how to retrieve it.

The scheme can be retrieved via `context.Scheme.Name`. It's not an ASOS-specific change, tho'.

If documentation is so important for you, would you be willing to financially support the effort?

**Comment by dozer75** [comment]
Thanks, I did look at the sample, but missed that since the code otherwise was quite similar. 

I don't need documentation, but a small migration guide would be nice when fundamentals are changed, like they did in ASP.NET Core 1.1 to 2.0. 

But now you gave me most of what I was needing, only thing left is to do logging like I did in UseOpenIdConnectServer, but that's out of scope for this issue and may be fixed in other ways.
